crates/core/src/operations/subscribe.rs

```diff
@@ -181,43 +181,41 @@ pub(crate) async fn request_subscribe(
             );
         }
 
+        // LOCAL-FIRST: If contract exists locally, complete subscription immediately.
```

Comment:
This does not make sense, when you subscribe you want to subscribe to other peers updates, not to your own updates.

We are conflating local contract subscriptions with network updates. Remember  they are different paths handled with different code paths and shouldn't be mixed up.